File: 剑指offer（c++）/和为S的连续正数序列.cpp
class Solution {
public:
    vector<vector<int> > FindContinuousSequence(int sum) {
        vector<vector<int>> allres;
        int a=1,b=2,i;
        while(a<b){
            int count=(a+b)*(b-a+1)/2;
            if(count<sum)
                b++;
            if(count==sum){
                vector<int> res;
                for(i=a;i<=b;i++)
                    res.push_back(i);
                allres.push_back(res);
                a++;
            }
            if(count>sum)
                a++;
        }
        return allres;
    }
};

File: 剑指offer（python）/和为S的连续正数序列.py
#- * -coding: utf - 8 - * -
class Solution:
  def FindContinuousSequence(self, tsum):
    #write code here res = []
	for i in range(1, tsum //2+1):
      sumRes=i
      for j in range(i+1,tsum//2+2):
        sumRes+=j
        if sumRes==tsum:
          res.append(list(range(i,j+1)))
          break
        elif sumRes>tsum:
          break
          return res